### PR TITLE
Include formDefinition in FormFieldDataResolver.get method

### PR DESCRIPTION
**Pull request by maarten-ritense**
Added 'get' to FormFieldDataResolver which includes the formDefinition JsonNode parameter, marked the other 'get' a Deprecated.

### CODE REVIEW

**Review by Tom-Ritense — APPROVED** [review]

File: contract/src/main/java/com/ritense/valtimo/contract/form/FormFieldDataResolver.java [hunk 1]
```diff
@@ -30,6 +32,8 @@ default boolean supports(String externalFormFieldType) {
         return supports(ExternalFormFieldType.fromKey(externalFormFieldType));
     }
 
+    @Deprecated(forRemoval = true, since = "9.21")
     Map<String, Object> get(String documentDefinitionName, UUID documentId, String... varNames);
 
+    Map<String, Object> get(String documentDefinitionName, UUID documentId, JsonNode formDefinition, String... varNames);
```
Comment (marijnritense) [inline contract/src/main/java/com/ritense/valtimo/contract/form/FormFieldDataResolver.java:38]: If you give this method a default implementation (e.g. just call the old get) it won't be a breaking change.

**Review by Tom-Ritense — APPROVED** [review]

File: contract/src/main/java/com/ritense/valtimo/contract/form/FormFieldDataResolver.java [hunk 2]
```diff
@@ -30,6 +30,15 @@ default boolean supports(String externalFormFieldType) {
         return supports(ExternalFormFieldType.fromKey(externalFormFieldType));
     }
 
-    Map<String, Object> get(String documentDefinitionName, UUID documentId, String... varNames);
+    @Deprecated(forRemoval = true, since = "9.21")
+    default Map<String, Object> get(String documentDefinitionName, UUID documentId, String... varNames) {
```
Comment (marijnritense) [inline contract/src/main/java/com/ritense/valtimo/contract/form/FormFieldDataResolver.java:34]: What I meant was for the new method to be default, not this one. So:

```
    default Map<String, Object> get(
        String documentDefinitionName,
        UUID documentId,
        FormFieldDataResolverProperties additionalProperties,
        String... varNames
    ) {
        return get(documentDefinitionName, documentId, varNames);
    }
```

Comment (Rick-Ritense) [reply contract/src/main/java/com/ritense/valtimo/contract/form/FormFieldDataResolver.java:34]: doesn't that make the code breaking when you change the default

I think you should change the default when the old method is removed in the future

Comment (ivo-ritense) [reply contract/src/main/java/com/ritense/valtimo/contract/form/FormFieldDataResolver.java:34]: The default would be to ensure backwards compatibility, the intent is still to override it in your implementation.

**Review by marijnritense — APPROVED** [review]

**Review by ivo-ritense — APPROVED** [review]

**Pull request merged**